include/zephyr/devicetree/map.h

```diff
@@ -0,0 +1,263 @@
+ *     int pspec_0 = DT_PHA_BY_IDX(DT_NODELABEL(n), gpio_map, 0, parent_specifier_0); // 2
+ *     int pspec_1 = DT_PHA_BY_IDX(DT_NODELABEL(n), gpio_map, 0, parent_specifier_1); // 3
+ * @endcode
+ *
+ * The map helper API also provides the following macros for convenient access to
+ * specific parts of a mapping entry:
+ *  - DT_MAP_CHILD_SPECIFIER_ARGS()
+ *  - DT_MAP_PARENT_SPECIFIER_ARGS()
+ *  - DT_MAP_PARENT_ARG()
+ *
+ * These macros extract, respectively, the child specifier arguments, the parent specifier
+ * arguments, and the parent node argument from a mapping element identified by its node ID,
+ * property name, and index.
+ *
+ * For instance:
+ *
+ * @code{.c}
+ *     #define SRC_AND_DST(node_id, prop, idx) \
+ *         { GET_ARG_N(1, DT_MAP_CHILD_SPECIFIER_ARGS(node_id, prop, idx)), \
+ *           GET_ARG_N(1, DT_MAP_PARENT_SPECIFIER_ARGS(node_id, prop, idx)) }
+ *
+ *     int src_and_dst[2][] = {
+ *         DT_FOREACH_PROP_ELEM_SEP(DT_NODELABEL(n), gpio_map, SRC_AND_DST, (,))
+ *     };
+ * @endcode
+ *
+ * The above expansion yields:
+ *
+ * @code{.c}
+ *     int src_and_dst[2][] = {{0, 2}, {4, 6}};
+ * @endcode
+ *
+ * @ingroup devicetree
+ * @{
+ */
+
+/**
+ * @brief Extracts a specified range of arguments.
+ *
+ * This helper macro first skips a given number of arguments and then selects
+ * the first @p len arguments from the remaining list.
+ *
+ * @param start The number of arguments to skip.
+ * @param len The number of arguments to extract after skipping.
+ * @param ... The list of input arguments.
+ */
+#define DT_MAP_HELPER_DO_ARGS_RANGE(start, len, ...)                                               \
+	GET_ARGS_FIRST_N(len, GET_ARGS_LESS_N(start, __VA_ARGS__))
+
+/**
+ * @brief Extracts a range of mapping arguments for a specific field.
+ *
+ * This macro concatenates the field name with the appropriate suffixes to determine
+ * the starting index and length of the arguments for a map entry, and then extracts
+ * those arguments.
+ *
+ * @param name The mapping field name (e.g., CHILD_SPECIFIER, PARENT).
+ * @param node_id The node identifier.
+ * @param prop The property name in lowercase and underscores.
+ * @param idx The index of the mapping entry.
+ * @param ... Additional arguments corresponding to the mapping entry.
+ */
+#define DT_MAP_HELPER_ARGS_RANGE(name, node_id, prop, idx, ...)                                    \
+	DT_MAP_HELPER_DO_ARGS_RANGE(DT_CAT3(DT_MAP_, name, _IDX)(node_id, prop, idx),              \
+				    DT_CAT3(DT_MAP_, name, _LEN)(node_id, prop, idx), __VA_ARGS__)
+
+/**
+ * @brief Retrieves the mapping entry at the specified index.
+ *
+ * @param node_id The node identifier.
+ * @param prop The property name in lowercase with underscores.
+ * @param idx The mapping entry index.
+ * @return The mapping entry as a list of comma-separated values.
+ */
+#define DT_MAP_IDX(node_id, prop, idx) DT_CAT5(node_id, _P_, prop, _MAP_IDX_, idx)
```

Comment:
```suggestion
#define DT_MAP_BY_IDX(node_id, prop, idx) DT_CAT5(node_id, _P_, prop, _MAP_IDX_, idx)
```

Should we continue to stick with the current style? At present, most APIs that take an `idx` parameter use the `_BY_IDX` suffix and also provide a simpler version of the API where `idx` defaults to 0.

Reply:
I have revised the whole code to comply with these conventions.